### PR TITLE
Slash Commands

### PR DESCRIPTION
**Pull request by asleepysheepy**
# Slash commands
Points our `package.json`file to discord.js' master branch in order to use interactions for slash commands and updates all existing commands to use slash commands

## Deploying

This PR only adds a very rudimentary system for deploying commands. There is a new envvar for `USERS_WHO_CAN_DEPLOY` which is an array of user snowflakes. Whenever a user whose snowflake in in that envvar sends the message `!deployWinnieCommands` commands will get registered with discord. Currently if the `NODE_ENV` envvar is set to `production` commands will be deployed globally, if not they are commands will be deployed to the guild in which the message was sent.

## Future updates needed

Currently there is no way to deploy localised commands. Ideally these will automatically update when the guild locale is changed but there is a complication there around how new commands get deployed to localised guilds and how discord handles global and guild commands with the same name

### CODE REVIEW

File: src/commands/utils/execute-top-level-command.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { CommandInteraction } from 'discord.js'
+import { GuildConfig } from '../../models'
+import { SubCommand } from '../../types/subcommand'
+import { Logger } from '../../core/logger'
+
+/**
+ * A genereic, reusable execute function for the top level command handler of commands
```
Comment (lisushka) [inline src/commands/utils/execute-top-level-command.ts:7]: genereic > generic

**Review by lisushka — CHANGES_REQUESTED** [review]
🙇🙇🙇

Apart from the one typo, this is good.  What's the plan for tests?

**Comment by asleepysheepy** [comment]
> What's the plan for tests?

What kind of tests?

**Comment by asleepysheepy** [comment]
> > What's the plan for tests?
> 
> What kind of tests?

Discussed offline, Decided there'd be too much mocking involved atm to be worth adding automated tests

**Review by lisushka — APPROVED** [review]
Ship it!

**Pull request merged**